### PR TITLE
Compile with target java 21, remove unsupported MC versions

### PR DESCRIPTION
**Pull request by SirYwell**
## Description

LinBus is compiled with Java 21, but older Spigot releases can't load those classes because the bundled ASM version is too old. This means we can't support these versions anymore, which also means we can target Java 21 with FAWE too.

1.20.1 still seems to be used somewhat widely according to bStats, but I don't see any way for us to still provide support for it.

```[tasklist]
### Submitter Checklist
- [x] Make sure you are opening from a topic branch (**/feature/fix/docs/ branch** (right side)) and not your main branch.
- [x] Ensure that the pull request title represents the desired changelog entry.
- [x] New public fields and methods are annotated with `@since TODO`.
- [x] I read and followed the [contribution guidelines](https://github.com/IntellectualSites/.github/blob/main/CONTRIBUTING.md).
```

### CODE REVIEW

**Comment by NotMyFault** [comment]
Oh I can't approve it by hey, lgtm

**Review by NotMyFault — APPROVED** [review]

**Review by dordsor21 — APPROVED** [review]

**Comment by dordsor21** [comment]
Should probably adjust the issue template

**Comment by MattBDev** [comment]
Sounds similar to how we had to split with Legacy FAWE due to the source code being a mess and other issues. I'm all for dropping support for older versions if there is no way around it. Could we bundle a more up to date version of ASM or would that cause some other compatibility issues?

**Comment by SirYwell** [comment]
> Could we bundle a more up to date version of ASM or would that cause some other compatibility issues?

The ASM version is provided and loaded by the server. We can't change that in any way.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -27,9 +27,9 @@ body:
       description: Which server version version you using? If your server version is not listed, it is not supported. Update to a supported version first.
       multiple: false
       options:
+        - '1.21'
         - '1.20.5/6'
         - '1.20'
```
Comment (dordsor21) [inline .github/ISSUE_TEMPLATE/bug_report.yml:32]: 1.20.2 I guess?

Comment (SirYwell) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: I changed it to mention only the versions we actually support now (unless I missed something)

**Pull request merged**